src/stream.rs

```diff
@@ -27,13 +27,13 @@ impl<S: AsyncRead + Unpin, T: AsyncRead + Unpin> AsyncRead for Stream<S, T> {
     ) -> Poll<std::io::Result<usize>> {
         #[project]
         match self.project() {
-            Stream::Plain(ref mut s) => Pin::new(s).poll_read(cx, buf),
-            Stream::Tls(ref mut s) => Pin::new(s).poll_read(cx, buf),
+            Stream::Plain(s) => s.poll_read(cx, buf),
+            Stream::Tls(s) => s.poll_read(cx, buf),
```

Comment:
As much as I do enjoy a bit simpler code here, I also find your thoughts on #90 reasonable, so I would probably prefer the approach from #90.